src/main/java/org/elmlang/intellijplugin/Elm.flex

```diff
@@ -41,15 +43,14 @@ RESERVED=("hiding" | "export" | "foreign" | "deriving")
 <IN_COMMENT> {
     "{-" {
         commentLevel++;
-        return COMMENT_CONTENT;
+        return START_COMMENT;
```

Comment:
I'm pretty sure that this change is going to create problems with nested, block comments.

For instance, in the 'add-import' intention, we need to scan for the last module doc comment to find the place to insert new imports. With the change you made on line 53, a nested block comment will cause that search to complete prematurely when it find an `END_COMMENT` but it's not actually the final `END_COMMENT`.

Maybe the right thing is to change the PsiTree (as you suggested in your PR notes) so that comments are grouped under a parent node instead of being inlined as siblings. In which case we can keep your change to the lexer so that start- and end-comment tokens are "visible", even when they are in a nested comment.